internal/librarian/commit_version_analyzer_test.go

```diff
@@ -135,9 +135,45 @@ func TestShouldInclude(t *testing.T) {
 	} {
 		t.Run(test.name, func(t *testing.T) {
 			t.Parallel()
-			got := shouldInclude(test.files, test.sourceRoots, test.excludePaths)
+			got := shouldIncludeForRelease(test.files, test.sourceRoots, test.excludePaths)
 			if got != test.want {
-				t.Errorf("shouldInclude(%v, %v, %v) = %v, want %v", test.files, test.sourceRoots, test.excludePaths, got, test.want)
+				t.Errorf("shouldIncludeForRelease(%v, %v, %v) = %v, want %v", test.files, test.sourceRoots, test.excludePaths, got, test.want)
+			}
+		})
+	}
+}
+
+func TestShouldIncludeForGeneration(t *testing.T) {
+	t.Parallel()
+	for _, test := range []struct {
+		name     string
+		files    []string
+		apiPaths []string
+		want     bool
+	}{
+		{
+			name:     "all_files_in_apiPaths",
+			files:    []string{"a/b/c.proto"},
+			apiPaths: []string{"a"},
+			want:     true,
+		},
+		{
+			name:     "some_files_in_apiPaths",
+			files:    []string{"a/b/c.proto", "e/f/g.proto"},
+			apiPaths: []string{"a"},
+			want:     true,
+		},
+		{
+			name:     "no_files_in_apiPaths",
```

Comment:
Could you add a test that specifies `source_roots` to verify the source roots are not affecting the result.

Reply:
Same as above, I think I can test this in `TestGetConventionalCommitsSinceLastGeneration`